Providers/Profile.swift

```diff
@@ -509,43 +533,117 @@ open class BrowserProfile: Profile {
-            if let tabsError = tabsResult.failureValue { return deferMaybe(tabsError) }
-
-            guard let clientRemoteTabs = tabsResult.successValue else { return
-                deferMaybe([])
+    // This function exists to service the `FxaPushMessengerHandler.handle` function and
+    // will be removed after the rust sync manager experiment is complete
+    public func getClient(fxaDeviceId: String) -> Deferred<Maybe<RemoteClient?>> {
+        if useRustSyncManager {
+            let msg = "Retrieving client records from rust tabs component"
+            logger.log(msg, level: .debug, category: .sync)
+
+            return tabs.getClient(fxaDeviceId: fxaDeviceId)
+        } else {
+            let msg = "Retrieving client records from BrowserDB clients table"
+            logger.log(msg, level: .debug, category: .sync)
+
+            return remoteClientsAndTabs.getClient(fxaDeviceId: fxaDeviceId)
+        }
+    }
+
+    // This function exists to service the `TabPeekViewController.setState` function and
+    // will be removed after the rust sync manager experiment is complete
+    public func getClientGUIDs(completion: @escaping (Set<GUID>) -> Void) {
+        if useRustSyncManager {
+            let msg = "Retrieving client GUIDs from rust tabs component"
+            logger.log(msg, level: .debug, category: .sync)
+
+            tabs.getClientGUIDs { (result, error) in
+                guard let guids = result else {
+                    return
+                }
+                completion(guids)
             }
+        } else {
+            let msg = "Retrieving client GUIDs from BrowserDB clients table"
+            logger.log(msg, level: .debug, category: .sync)
 
-            return self.remoteClientsAndTabs.getClients().bind { result in
-                if let error = result.failureValue { return deferMaybe(error) }
+            remoteClientsAndTabs.getClientGUIDs().upon { result in
+                guard let guids = result.successValue else {
+                    return
+                }
+                completion(guids)
+            }
+        }
+    }
 
-                guard let clients = result.successValue else { return deferMaybe([]) }
+    func getRustTabsWithClients() -> Deferred<Maybe<[ClientAndTabs]>> {
+        logger.log("Getting all tabs and clients", level: .debug, category: .tabs)
 
-                let clientAndTabs: [ClientAndTabs] = clientRemoteTabs.map { record in
-                    // We check if the application services clientId matches any client
-                    // GUID. If a client is found we return a record, otherwise we
-                    // continue to the next application services record.
-                    let localClient = clients.first(where: { $0.guid == record.clientId })
+        guard let accountManager = self.rustFxA.accountManager.peek(),
+              let state = accountManager.deviceConstellation()?.state() else {
+            return deferMaybe([])
```

Comment:
nit: preserving our guard style

```suggestion
        guard let accountManager = self.rustFxA.accountManager.peek(),
              let state = accountManager.deviceConstellation()?.state()
        else {
            return deferMaybe([])
```

Reply:
@adudenamedruby keeping the top as the suggestion looks really odd

Reply:
Its git showing weirdness adding this in